### ISSUE TITLE
Support Configuration Caching

### ISSUE THREAD

**Issue by runningcode** [gradle/android-cache-fix-gradle-plugin#101]
Starting this issue to strategize about this:

Are we going to need to compile against Gradle 6.0 to take advantage of newer APIs?
Should we drop support for AGP 3.5? No new cache fixes against AGP 3.5 are likely going to be developed, maybe we can just suggest AGP 3.5 users stay with version 1.0.7?

Are we going to be able to do all the funny business we do with task manipulation with configuration caching?

**Comment by ZacSweers** [comment]
```
org.gradle.api.InvalidUserCodeException: Read system property 'org.gradle.android.cache-fix.ignoreVersionCheck'
	at org.gradle.instantexecution.SystemPropertyAccessListener.systemPropertyQueried(SystemPropertyAccessListener.kt:92)
	at org.gradle.internal.classpath.Instrumented.getBoolean(Instrumented.java:133)
	at org.gradle.android.AndroidCacheFixPlugin.isSupportedAndroidVersion(AndroidCacheFixPlugin.groovy:53)
	at org.gradle.android.AndroidCacheFixPlugin.apply(AndroidCacheFixPlugin.groovy:63)
	at org.gradle.android.AndroidCacheFixPlugin.apply(AndroidCacheFixPlugin.groovy)
```

```
org.gradle.api.InvalidUserCodeException: Read system property 'org.gradle.android.cache-fix.ignoreVersionCheck'
	at org.gradle.instantexecution.SystemPropertyAccessListener.systemPropertyQueried(SystemPropertyAccessListener.kt:92)
	at org.gradle.internal.classpath.Instrumented.getBoolean(Instrumented.java:133)
	at org.gradle.android.AndroidCacheFixPlugin.isSupportedAndroidVersion(AndroidCacheFixPlugin.groovy:53)
	at org.gradle.android.AndroidCacheFixPlugin.apply(AndroidCacheFixPlugin.groovy:63)
	at org.gradle.android.AndroidCacheFixPlugin.apply(AndroidCacheFixPlugin.groovy)
```

**Comment by ZacSweers** [comment]
```
at org.gradle.android.workarounds.CompilerArgsProcessor$_ensureApplied_closure1.doCall(CompilerArgsProcessor.groovy:44)
```

**Comment by ZacSweers** [comment]
```
at org.gradle.android.workarounds.RoomSchemaLocationWorkaround$_apply_closure6$_closure17.doCall(RoomSchemaLocationWorkaround.groovy:124)
```

**Comment by CristianGM** [comment]
```
What went wrong:
Execution failed for task ':libraries:Design:compileDebugLibraryResources'.
> Could not find method setPropertyValue() for arguments [task ':libraries:Design:compileDebugLibraryResources', null] on task ':libraries:Design:compileDebugLibraryResources' of type com.android.build.gradle.tasks.CompileLibraryResourcesTask.
```

because: https://github.com/gradle/android-cache-fix-gradle-plugin/blob/17ff9dec47b935207f469cee9163f6e6a1316b10/src/main/groovy/org/gradle/android/workarounds/CompileLibraryResourcesWorkaround.groovy

**Comment by kozaxinan** [comment]
With Gradle 6.8-milestone-1
```
org.gradle.api.InvalidUserCodeException: Listener registration 'TaskExecutionGraph.beforeTask' by org.gradle.execution.taskgraph.DefaultTaskExecutionGraph@7f8bf9f4 is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onBuildScopeListenerRegistration(ConfigurationCacheProblemsListener.kt:90)
```
And https://github.com/gradle/android-cache-fix-gradle-plugin/issues/101#issuecomment-670975356 still exists.

**Comment by ghale** [comment]
The remaining configuration cache issues should be fixed by https://github.com/gradle/android-cache-fix-gradle-plugin/pull/143 and are in the 2.1.0 release.
